nio_cli/commands/blockcheck.py

```diff
@@ -0,0 +1,241 @@
+import json
+import os
+import sys
+import subprocess
+
+from .base import Base
+
+
+class BlockCheck(Base):
+
+    # This should be run from inside the block dir
+
+    def __init__(self, options, *args, **kwargs):
+
+        super().__init__(options, *args, **kwargs)
+        self._block = os.getcwd().split('/')[-1]
+        self.all_contents = os.listdir('.')
+        self.block_files_including_base = [
+            f for f in self.all_contents if '.py' in f
+        ]
+        self.block_files = [
+            f for f in self.all_contents if 'block.py' in f and 'base' not in f
+        ]
+        self.blocks_in_spec = []
+        self.spec_versions_dict = {}
+        self.release_versions_dict = {}
+        self.file_versions_and_names_dict = self.get_versions_and_class_names()
+
+    def get_versions_and_class_names(self):
+
+        name_version_dict = {
+            'classes': [],
+            'versions': [],
+        }
+        for block in self.block_files_including_base:
+            with open(block) as f:
+                lines = [l.rstrip() for l in f.readlines()]
+                for line in lines:
+                    if 'class ' in line:
+                        split1 = line.split(' ')[1]
+                        class_name = split1.split('(')[0]
+                        name_version_dict['classes'].append(class_name)
+                    if '= VersionProperty' in line:
+                        replace1 = line.replace('"', '^')
+                        replace2 = replace1.replace("'", '^')
+                        version_string = replace2.split("^")[1]
+                        name_version_dict['versions'].append(version_string)
+        return name_version_dict
+
+    def run(self):
+
+        self.print_check('PEP8')
+        self.check_pep8()
+
+        self.print_check('spec.json')
+        self.check_spec()
+
+        self.print_check('README.md')
+        self.check_readme()
+
+        self.print_check('release.json')
+        self.check_release()
+
+        self.print_check('version')
+        self.check_version()
+
+        self.print_check('class and file name')
+        self.check_naming()
+
+    def check_pep8(self):
+
+        shell_pep8 = 'pep8 .'
+        subprocess.call(shell_pep8, shell=True)
+        print('')
+
+    def check_spec(self):
+
+        if os.path.exists('spec.json'):
+            with open('spec.json') as f:
+                try:
+                    spec_dict = json.load(f)
+                except json.JSONDecodeError:
+                    print('spec.json file is either incomplete or '
+                          'has an invalid JSON object')
+                    print(
+                        '\n**Run `nio buildspec {}` '
+                        'from the project directory '
+                        'and re-run this check**\n'.format(self._block)
+                    )
+                    sys.exit()
```

Comment:
We only exit after each check for the spec file.  That's because `self.blocks_in_spec` is used as the base for the other checks, so if that list is incorrect, the other checks will print out errors that may be fixed by running `nio buildspec [repo name]`